### PR TITLE
Test Kitchen Config Refactor

### PR DESCRIPTION
**Pull request by zwall-bp**
All tests green.

There are some RAM attributes which seems to be dependent on your machine's specs, so there may be verification issues on your machine for Vagrant and Dokken.

### CODE REVIEW

File: test/integration/server/inspec/server_spec.rb [hunk 1]
```diff
@@ -52,12 +55,11 @@
...
   its('mysqld.log_bin_trust_function_creators') { should eq '1' }
-  its('mysqld.innodb_large_prefix') { should eq 'true' } if os.release.to_i < 8 # Deprecated in mysql 5.7
   its('mysqld.userstat') { should eq 'true' }
-  its('mysqld.innodb_file_format') { should eq 'barracuda' } if os.release.to_i < 8 # Deprecated in mysql 5.7
   its('content') { should match(/^innodb_file_per_table$/) }
-  its('mysqld.innodb_buffer_pool_size') { should eq '2652M' } if os.release.to_i < 8
-  its('mysqld.innodb_buffer_pool_size') { should eq '2611M' } if os.release.to_i >= 8 && os.name == 'almalinux'
+  its('mysqld.innodb_buffer_pool_size') { should eq '2557M' } unless vagrant || docker
+  its('mysqld.innodb_buffer_pool_size') { should eq '1236M' } if vagrant
+  its('mysqld.innodb_buffer_pool_size') { should eq '11141M' } if docker
```
Comment (ramereth) [inline test/integration/server/inspec/server_spec.rb:62]: This test is going to be different based on which workstation you're running this on. For now I'd just skip this for now.

File: test/integration/server/inspec/server_spec.rb [hunk 2]
```diff
@@ -52,12 +55,11 @@
 
 describe mysql_conf('/etc/my.cnf') do
   its('mysqld.log_bin_trust_function_creators') { should eq '1' }
-  its('mysqld.innodb_large_prefix') { should eq 'true' } if os.release.to_i < 8 # Deprecated in mysql 5.7
   its('mysqld.userstat') { should eq 'true' }
-  its('mysqld.innodb_file_format') { should eq 'barracuda' } if os.release.to_i < 8 # Deprecated in mysql 5.7
   its('content') { should match(/^innodb_file_per_table$/) }
```
Comment (ramereth) [inline test/integration/server/inspec/server_spec.rb:59]: This looks like CentOS 7 removals. Let's leave that for a separate PR for now and keep this as-is for now.

File: test/integration/server/inspec/server_spec.rb [hunk 3]
```diff
@@ -71,8 +73,9 @@
 end
 
 describe kernel_parameter('vm.min_free_kbytes') do
-  its('value') { should eq 38799 } if os.release.to_i < 8
-  its('value') { should eq 38205 } if os.release.to_i >= 8 && os.name == 'almalinux'
+  its('value') { should eq 37406 } unless vagrant || docker
+  its('value') { should eq 18083 } if vagrant
+  its('value') { should eq 162990 } if docker
```
Comment (ramereth) [inline test/integration/server/inspec/server_spec.rb:78]: Skip this ass well for the same reason above.

**Review by ramereth — CHANGES_REQUESTED** [review]

**Comment by ramereth** [comment]
@zwall-bp please rebase and resolve the conflicts.

File: test/integration/server/inspec/server_spec.rb [hunk 4]
```diff
@@ -46,9 +49,11 @@
 
 describe mysql_conf('/etc/my.cnf') do
   its('mysqld.log_bin_trust_function_creators') { should eq '1' }
+  its('mysqld.innodb_large_prefix') { should eq 'true' } if os.release.to_i < 8 # Deprecated in mysql 5.7
   its('mysqld.userstat') { should eq 'true' }
+  its('mysqld.innodb_file_format') { should eq 'barracuda' } if os.release.to_i < 8 # Deprecated in mysql 5.7
```
Comment (ramereth) [inline test/integration/server/inspec/server_spec.rb:54]: I removed these in #85, please revert this change.

**Review by ramereth — CHANGES_REQUESTED** [review]

File: test/integration/server/inspec/server_spec.rb (hunk 4 above)
Comment (zwall-bp) [reply test/integration/server/inspec/server_spec.rb]: oop
